e2e/test.js

```diff
@@ -0,0 +1,49 @@
+const expect = require('chai').expect;
+
+describe('InstantSearch e-commerce demo', () => {
+  before(() => {
+    browser.maximizeWindow();
+  });
+
+  it('must work', async () => {
+    await browser.url('http://localhost:5000/examples/e-commerce/');
+
+    // Select "Apple" brand in list
+    const brand = await browser.$('span=Apple');
+    await brand.click();
+
+    // Fill search input with "macbook"
+    const searchInput = await browser.$('[type=search]');
+    await searchInput.setValue('macbook');
+
+    // Wait for the results list to be updated (wait for the "macbook" word to be highlighted)
+    await browser.waitUntil(
+      async () => (await browser.$$('mark=MacBook')).length > 0,
+      10000
+    );
+
+    // Get title for each result
+    const hits = await browser.$$('.hit h1');
+    const hitsText = await Promise.all(hits.map(hit => hit.getText()));
+
+    // Compare them to expected titles
+    expect(hitsText).to.deep.equal([
+      'Apple - MacBook Air® (Latest Model) - 13.3" Display - Intel Core i5 - 8GB Memory - 128GB Flash Storage - Silver',
```

Comment:
Here I think what we expect is to have Apple in all the hits text so maybe we should just check that?
There are less changes the test will fail if for some reason the ranking a or hit changed.

Reply:
Mmm, I prefer to test that we get exactly the results we expect, and checking for only "Apple" is too vague IMO.

I find this as a good thing to have the test to fail if the ranking or a hit change :innocent:

Reply:
This is something we can do with a dedicated demo: we can show the value of the attribute that will be filtered in the UI so we can test it

Reply:
That would be equivalent to exposing some internals for testing, not a good practice and it would ruin the purpose of having end-2-end tests.

Reply:
Not really true, the attribute is what proves that it's filtered, since the filter isn't related to the search query per se, and can be in multiple attributes

Reply:
Let's not forget that we are supposed to test from a end user POV, so what matter at the end is that the result list is correct.